File: lesson_1/mario_less.c
#include <cs50.h>
#include <stdio.h>

void draw(int n);

int main(void)
{
    int input;
    do 
    {
        input = get_int("Height: ");
    }
    while (input < 1 || input > 8);
    draw(input);
}

void draw(int n) 
{
    if (n == 1)
    {
        printf("#\n");
    }
    else 
    {
        int count = n - 1;
        for (int i = 0; i < n; i++) 
        {
            for (int j = 0; j < n; j++) 
            {
                if (j < count)
                { 
                    printf(" ");
                }
                else
                { 
                    printf("#");
                }
            }
            printf("\n");
            count--;
        }
    }
}

File: pset6/mario-less.py
# Prints out a half-pyramid of a specified height

from cs50 import get_int

# Adapting my old C code to Python

# Receiving input from user, will repeat until the received value is a value between 1 and 8
while True:
    size = get_int("Height: ")
    # Range can receive an optional starting value to the left of the end value
    if size in range(1, 9):
        break

# Printing each floor and adding whitespace when required
count = size - 1
for i in range(size):
    for j in range(size):
        if j < count:
            print(" ", end="")
        else:
            print("#", end="")
    print()
    count -= 1